modules/salmon/1.1/salmon.smk

```diff
@@ -15,6 +15,25 @@
 # Import package with useful functions for developing analysis modules
 import oncopipe as op
 import pandas as pd
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
 # Setup module and store module-specific configuration in `CFG`
 # `CFG` is a shortcut to `config["lcr-modules"]["salmon"]`
```

Comment:
Best to avoid acronyms where possible and instead use informative variable names.

Reply:
Not my choice, it is the current standard.

Reply:
`CFG` was used because it appears dozens of times in the Snakefiles (sometimes multiple times in one line), so it being more explicit (which I prefer in general) would make the lines too long. The best analogy I could come up with is `cd` vs `change_directory`. I'm glad they went with just `cd`.

Reply:
Thanks for the clarification Bruno!